projects/cardiovascular-patient-app/src/app/last-report-bp/last-report-bp.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component } from '@angular/core';
+import { DataLayerManagerService, formatDate, formatTime } from 'ngx-charts-on-fhir';
+import { map } from 'rxjs';
+
+interface LastReportedBPdata {
+  systolic: { date: string; value: number };
+  diastolic: { date: string; value: number };
+}
+
+@Component({
+  selector: 'last-report-bp',
+  templateUrl: './last-report-bp.component.html',
+  styleUrls: ['./last-report-bp.component.css'],
+})
+export class LastReportBPComponent {
+  lastReportedBPdata?: LastReportedBPdata;
+
+  constructor(private layerManager: DataLayerManagerService) { }
+
+  ngOnInit(): void {
+    this.layerManager.allLayers$
+      .pipe(
+        map((layers) => {
+          const bloodPressureLayer = layers.find((layer) => layer.name === 'Blood Pressure');
+          return bloodPressureLayer?.datasets.map((data) => data.data.slice(-1)[0]);
+        }
+        )
+      )
+      .subscribe((layers: any) => {
+        if (layers?.length > 2) {
+          this.lastReportedBPdata = {
+            systolic: { date: `${formatDate(layers[3].x)} at ${formatTime(layers[3].x)}`, value: layers[3].y },
+            diastolic: { date: `${formatDate(layers[2].x)} at ${formatTime(layers[2].x)}`, value: layers[2].y },
+          };
+        }
```

Comment:
1. We should not rely on dataset indexes because the order might change.
2. We want to show the most recent observation, regardless of whether it was home or office.

Maybe something like:
```ts
datasets.filter(dataset => dataset.label.startsWith('Systolic'))
        .map(dataset => dataset.data.at(-1)) // most recent in each dataset
        .sort((a,b)=>b.x - a.x)
        .at(-1) // most recent overall
```
Or you could try lodash groupBy/mapValues instead of filter/map if you want to process both systolic/diastolic without code duplication.

Reply:
Okay, @semicolin I will try with lodash.